docs/how-trace-inputlocator-works.rst

```diff
@@ -0,0 +1,98 @@
+:orphan:
+
+Metadata acquisition
+====================
+
+trace-inputlocator
+------------------
+The CEA contains a tool for the acquisition of metadata specific to each scenario, called via the command line interface
+by `cea trace-inputlocator --scripts <script_name>`. As the name suggests, the trace-inputlocator module records which methods are
+called from `inputlocator.py` during script execution. Having established which script calls which inputlocator method (and subsequently which file),
+the schema for each file is then read. Stored as a yml file within the directory of the current scenario, the keys are as follows:
+
+- 'locator_method':
+    - 'description' : the doc string associated with the inputlocator method
+    - 'created_by' : list of scripts which are responsible for writing the file (empty list if input file)
+    - 'used_by' : list of scripts which read the file (empty list if no dependencies exist)
+    - 'file_path' : the absolute path of the file
+    - 'file_type' : the file extension (e.g. 'xls' or 'dbf')
+    - 'schema' :
+        - variable or sheet name (for xls files)
+            - 'sample_data' : the last non-null entry of the file
+            - 'types_found' : list of python types encountered when reading the file, including None types
+
+Note! The first time metadata is generated, the order of the scripts is important as the 'created_by' list is updated each time a script is run. For example,
```

Comment:
I'm surprised by this. Can we meet tomorrow to discuss this?

Reply:
what is surprising about it? if the data is already there but the trace-inputlocator has not been run for a dependency, then how is it supposed to know which script created the data?

Reply:
well... the Note! sounds like it's a problem, but the explanation suggests it just works... and would just update the `created_by` when it get's there?